### PR TITLE
NavBar ReDesign finishing touches + ESLint Change

### PR DESCRIPTION
**Pull request by ChaseSizemore**
- Changed <a> tag to <Link>
- Removed ESLint rule to allow use of  tag
- prevented scrolling when mobile menu is open

### CODE REVIEW

**Comment by ChaseSizemore** [comment]
I made this PR with the hope that it passes the CI/CD this time when the first instance did not because of the ESLint rule

File: components/Header.tsx [hunk 1]
```diff
@@ -28,6 +29,19 @@ export default function Header() {
   /**
    * Prevent scrolling when the mobile menu is open
    */
+  useEffect(() => {
```
Comment (junaid-ahmed7) [inline components/Header.tsx:32]: whats this useEffect doing?

File: components/Header.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ export default function Header() {
       if (window.innerWidth >= 768) {
         // Tailwind's md breakpoint
         setIsMenuOpen(false);
+        document.body.classList.remove('menu-open'); // Remove the CSS class when the menu is closed
```
Comment (junaid-ahmed7) [inline components/Header.tsx:22]: is this needed? `setIsMenuOpen` should trigger our useEffect which should remove the class

Comment (ChaseSizemore) [reply components/Header.tsx:22]: Totally correct. Forgot to move the original implementation as it stopped working. I'll make that change.

**Review by junaid-ahmed7 — APPROVED** [review]
looks great - hopefully passes the CI/CD checks

**Pull request merged**